extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/ListChunkWriter.java

```diff
@@ -143,16 +143,23 @@ private ListChunkInputStream(
 
                 final RowSetBuilderSequential innerSubsetBuilder = RowSetFactory.builderSequential();
                 final RowSet toUse = mySubset == null ? RowSetFactory.flat(context.getChunk().size()) : mySubset;
-                toUse.forAllRowKeys(key -> {
-                    int startOffset = context.offsets.get(LongSizedDataStructure.intSize(DEBUG_NAME, key));
-                    int endOffset = context.offsets.get(LongSizedDataStructure.intSize(DEBUG_NAME, key + 1));
-                    if (limit < 0) {
-                        startOffset = Math.max(startOffset, endOffset + limit);
-                    } else if (limit > 0) {
-                        endOffset = Math.min(endOffset, startOffset + limit);
-                    }
+                toUse.forAllRowKeys(lkey -> {
+                    int key = LongSizedDataStructure.intSize(DEBUG_NAME, lkey);
```

Comment:
If we are worried about overflow, do we need to do an lkey + 1 to avoid wraparound on line 153?  Or more worryingly the multiplications on line 161?

Perhaps instead we should do that in long space and then check it?